crates/tiny/src/main.rs

```diff
@@ -178,3 +193,41 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth) -> Result<SASLAuth, String> {
+    match sasl_config {
+        config::SASLAuth::Plain { username, password } => Ok(SASLAuth::Plain {
+            username: username.to_string(),
+            password: password.to_string(),
+        }),
+        config::SASLAuth::External { pem } => {
+            // TLS must be on for EXTERNAL
+            if !tls {
+                Err("TLS not enabled".to_string())
```

Comment:
Since we show this error message in the mentions tab, it should mention the server name, otherwise it may be difficult for the user to see where it's coming from and which server config needs to be updated. Maybe something like `format!("SASL EXTERNAL authentication only works with TLS connections. Enable TLS for {} in the config file.", server_name)`.

Just realized while typing this command, we can show errors there in the server tabs instead of mentions tab. Doesn't hurt to improve this message though. Maybe something like `"TLS is not enabled for this server, but SASL EXTERNAL authentication requires SASL. Please enable SASL for this server in the config file."` (feel free to reword).